orchagent/vxlanorch.cpp

```diff
@@ -312,3 +547,135 @@ bool VxlanTunnelMapOrch::delOperation(const Request& request)
 
     return true;
 }
+
+sai_object_id_t
+VxlanVrfMapOrch::createNextHopTunnel(const std::string& vName, IpAddress& ipAddr, MacAddress macAddress, uint32_t vni)
+{
+    if (vxlan_vrf_tunnel_.find(vName) ==  vxlan_vrf_tunnel_.end())
+    {
+        return SAI_NULL_OBJECT_ID;
+    }
+
+    sai_ip_address_t host_ip;
+    swss::copy(host_ip, ipAddr);
+
+    sai_mac_t mac, *macptr = nullptr;
+    if (macAddress)
+    {
+        memcpy(mac, macAddress.getMac(), ETHER_ADDR_LEN);
+        macptr = &mac;
+    }
+
+    sai_object_id_t nh_id, tunnel_id = vxlan_vrf_tunnel_[vName];
+
+    if (create_nexthop_tunnel(host_ip, vni, macptr, tunnel_id, &nh_id) != SAI_STATUS_SUCCESS)
+    {
+        throw std::runtime_error("Can't create nexthop tunnel object");
+    }
+
+    SWSS_LOG_INFO("NH tunnel created for %s, id 0x%lx", vName.c_str(), nh_id);
+    return nh_id;
+}
+
+void VxlanVrfMapOrch::vnetTaskHandler(string& vrf_name, VxlanTunnel_T& tunnel_obj, handler_pair& cap_obj)
+{
+    VNetOrch* vnet_orch = gDirectory.get<VNetOrch*>();
+    if (!tunnel_obj->isActive())
+    {
+        if (vnet_orch->isVnetExecVrf())
+        {
+            tunnel_obj->createTunnel(MAP_T::VRID_TO_VNI, MAP_T::VNI_TO_VRID);
+        }
+        else
+        {
+            tunnel_obj->createTunnel(MAP_T::BRIDGE_TO_VNI, MAP_T::VNI_TO_BRIDGE);
+        }
+    }
+
+    cap_obj.first = vnet_orch->getEncapMapId(vrf_name);
+    cap_obj.second = vnet_orch->getDecapMapId(vrf_name);
+}
+
+bool VxlanVrfMapOrch::addOperation(const Request& request)
+{
+    SWSS_LOG_ENTER();
+
+    auto tunnel_name = request.getKeyString(0);
+    VxlanTunnelOrch* tunnel_orch = gDirectory.get<VxlanTunnelOrch*>();
+    if (!tunnel_orch->isTunnelExists(tunnel_name))
+    {
+        SWSS_LOG_WARN("Vxlan tunnel '%s' doesn't exist", tunnel_name.c_str());
+        return false;
+    }
+
+    auto vni_id  = static_cast<sai_uint32_t>(request.getAttrUint("vni"));
+    if (vni_id >= 1<<24)
+    {
+        SWSS_LOG_ERROR("Vxlan vni id is too big: %d", vni_id);
+        return true;
```

Comment:
>true [](start = 15, length = 4)

why return true here?

Reply:
Returning true to remove the entry from msgq to avoid further retries.